encode_test.go

```diff
@@ -1739,3 +1739,134 @@ values:
 		t.Fatalf("failed to encode: got = %s", string(b))
 	}
 }
+
+type issue401 struct{}
+
+func (b *issue401) MarshalYAML() ([]byte, error) {
+	v, err := yaml.Marshal("test")
+	if err != nil {
+		return nil, err
+	}
+	return []byte(fmt.Sprintf("%s %s", "!!timestamp", string(v))), nil
+}
+
+func TestBytesMarshalerWithTag(t *testing.T) {
```

Comment:
This test case does not validate the behavior when a tag is used for the map value. Since TagNode must be used during encoding, we need to first parse the YAML string with tags into an `ast.File` using the parser package ( `parser.ParseBytes()` ), and then convert it back to a string for validation

Reply:
I agree, I replaced the tests!